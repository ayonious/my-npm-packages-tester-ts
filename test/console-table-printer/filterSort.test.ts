import { expect } from '@jest/globals';
import { Table } from 'console-table-printer';

describe('Table with data manipulation', () => {
  // Sample data for all tests
  const sampleData = [
    { id: 1, name: 'Alice', department: 'Engineering', salary: 95000 },
    { id: 2, name: 'Bob', department: 'Marketing', salary: 78000 },
    { id: 3, name: 'Charlie', department: 'Engineering', salary: 92000 },
    { id: 4, name: 'Diana', department: 'HR', salary: 67000 },
    { id: 5, name: 'Edward', department: 'Marketing', salary: 82000 },
    { id: 6, name: 'Fiona', department: 'Engineering', salary: 99000 },
  ];

  it('should display filtered data (Engineering department only)', () => {
    const table = new Table({
      title: 'Engineering Department',
      columns: [
        { name: 'id', alignment: 'left' },
        { name: 'name', alignment: 'left' },
        { name: 'department', alignment: 'left' },
        { name: 'salary', alignment: 'right' },
      ],
    });

    // Filter data before adding to table
    const engineeringOnly = sampleData.filter(
      (row) => row.department === 'Engineering'
    );
    table.addRows(engineeringOnly);

    expect(table.render()).toMatchSnapshot();
  });

  it('should display sorted data (salary ascending)', () => {
    const table = new Table({
      title: 'Employees Sorted by Salary (Ascending)',
      columns: [
        { name: 'id', alignment: 'left' },
        { name: 'name', alignment: 'left' },
        { name: 'department', alignment: 'left' },
        { name: 'salary', alignment: 'right' },
      ],
    });

    // Sort data before adding to table
    const sortedBySalary = [...sampleData].sort((a, b) => a.salary - b.salary);
    table.addRows(sortedBySalary);

    expect(table.render()).toMatchSnapshot();
  });

  it('should display sorted data (salary descending)', () => {
    const table = new Table({
      title: 'Employees Sorted by Salary (Descending)',
      columns: [
        { name: 'id', alignment: 'left' },
        { name: 'name', alignment: 'left' },
        { name: 'department', alignment: 'left' },
        { name: 'salary', alignment: 'right' },
      ],
    });

    // Sort data before adding to table
    const sortedBySalaryDesc = [...sampleData].sort(
      (a, b) => b.salary - a.salary
    );
    table.addRows(sortedBySalaryDesc);

    expect(table.render()).toMatchSnapshot();
  });

  it('should display filtered and sorted data (Marketing department, sorted by salary)', () => {
    const table = new Table({
      title: 'Marketing Department (Sorted by Salary)',
      columns: [
        { name: 'id', alignment: 'left' },
        { name: 'name', alignment: 'left' },
        { name: 'department', alignment: 'left' },
        { name: 'salary', alignment: 'right' },
      ],
    });

    // Filter and sort data before adding to table
    const marketingOnly = sampleData
      .filter((row) => row.department === 'Marketing')
      .sort((a, b) => b.salary - a.salary);

    table.addRows(marketingOnly);

    expect(table.render()).toMatchSnapshot();
  });
});
