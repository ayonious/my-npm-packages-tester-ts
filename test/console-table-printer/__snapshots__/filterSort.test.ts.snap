// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Table with data manipulation should display filtered and sorted data (Marketing department, sorted by salary) 1`] = `
"[01m   Marketing Department (Sorted by   
               Salary)               [0m
┌────┬────────┬────────────┬────────┐
│[37m [0m[01mid[0m │[37m [0m[01mname  [0m │[37m [0m[01mdepartment[0m │[37m [0m[01msalary[0m │
├────┼────────┼────────────┼────────┤
│[37m [0m[37m5 [0m │[37m [0m[37mEdward[0m │[37m [0m[37mMarketing [0m │[37m [0m[37m 82000[0m │
│[37m [0m[37m2 [0m │[37m [0m[37mBob   [0m │[37m [0m[37mMarketing [0m │[37m [0m[37m 78000[0m │
└────┴────────┴────────────┴────────┘"
`;

exports[`Table with data manipulation should display filtered data (Engineering department only) 1`] = `
"[01m        Engineering Department         [0m
┌────┬─────────┬─────────────┬────────┐
│[37m [0m[01mid[0m │[37m [0m[01mname   [0m │[37m [0m[01mdepartment [0m │[37m [0m[01msalary[0m │
├────┼─────────┼─────────────┼────────┤
│[37m [0m[37m1 [0m │[37m [0m[37mAlice  [0m │[37m [0m[37mEngineering[0m │[37m [0m[37m 95000[0m │
│[37m [0m[37m3 [0m │[37m [0m[37mCharlie[0m │[37m [0m[37mEngineering[0m │[37m [0m[37m 92000[0m │
│[37m [0m[37m6 [0m │[37m [0m[37mFiona  [0m │[37m [0m[37mEngineering[0m │[37m [0m[37m 99000[0m │
└────┴─────────┴─────────────┴────────┘"
`;

exports[`Table with data manipulation should display sorted data (salary ascending) 1`] = `
"[01mEmployees Sorted by Salary (Ascending) [0m
┌────┬─────────┬─────────────┬────────┐
│[37m [0m[01mid[0m │[37m [0m[01mname   [0m │[37m [0m[01mdepartment [0m │[37m [0m[01msalary[0m │
├────┼─────────┼─────────────┼────────┤
│[37m [0m[37m4 [0m │[37m [0m[37mDiana  [0m │[37m [0m[37mHR         [0m │[37m [0m[37m 67000[0m │
│[37m [0m[37m2 [0m │[37m [0m[37mBob    [0m │[37m [0m[37mMarketing  [0m │[37m [0m[37m 78000[0m │
│[37m [0m[37m5 [0m │[37m [0m[37mEdward [0m │[37m [0m[37mMarketing  [0m │[37m [0m[37m 82000[0m │
│[37m [0m[37m3 [0m │[37m [0m[37mCharlie[0m │[37m [0m[37mEngineering[0m │[37m [0m[37m 92000[0m │
│[37m [0m[37m1 [0m │[37m [0m[37mAlice  [0m │[37m [0m[37mEngineering[0m │[37m [0m[37m 95000[0m │
│[37m [0m[37m6 [0m │[37m [0m[37mFiona  [0m │[37m [0m[37mEngineering[0m │[37m [0m[37m 99000[0m │
└────┴─────────┴─────────────┴────────┘"
`;

exports[`Table with data manipulation should display sorted data (salary descending) 1`] = `
"[01mEmployees Sorted by Salary (Descending)[0m
┌────┬─────────┬─────────────┬────────┐
│[37m [0m[01mid[0m │[37m [0m[01mname   [0m │[37m [0m[01mdepartment [0m │[37m [0m[01msalary[0m │
├────┼─────────┼─────────────┼────────┤
│[37m [0m[37m6 [0m │[37m [0m[37mFiona  [0m │[37m [0m[37mEngineering[0m │[37m [0m[37m 99000[0m │
│[37m [0m[37m1 [0m │[37m [0m[37mAlice  [0m │[37m [0m[37mEngineering[0m │[37m [0m[37m 95000[0m │
│[37m [0m[37m3 [0m │[37m [0m[37mCharlie[0m │[37m [0m[37mEngineering[0m │[37m [0m[37m 92000[0m │
│[37m [0m[37m5 [0m │[37m [0m[37mEdward [0m │[37m [0m[37mMarketing  [0m │[37m [0m[37m 82000[0m │
│[37m [0m[37m2 [0m │[37m [0m[37mBob    [0m │[37m [0m[37mMarketing  [0m │[37m [0m[37m 78000[0m │
│[37m [0m[37m4 [0m │[37m [0m[37mDiana  [0m │[37m [0m[37mHR         [0m │[37m [0m[37m 67000[0m │
└────┴─────────┴─────────────┴────────┘"
`;
